controller/api/destination/server.go

```diff
@@ -343,7 +354,7 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 // sendEndpointProfile sends a DestinationProfile response back to the client.
 // If the pod argument is provided, the profile sent to the client will
 // include an endpoint. Otherwise, the default profile is sent.
-func (s *server) sendEndpointProfile(stream pb.Destination_GetProfileServer, pod *corev1.Pod, port uint32) error {
+func (s *server) sendEndpointProfile(stream pb.Destination_GetProfileServer, pod *corev1.Pod, port uint32) (*profileTranslator, error) {
```

Comment:
it's a bit confusing that a function called `sendEndpointProfile` returns a profileTranslator for future updates.  Perhaps this would be more clear if this function and serverAdaptor were rolled together to create a struct called `endpointProfileTranslator`.  `endpointProfileTranslator` would take a GetProfileServer stream, a pod, and a port as constructor arguments and would implement the ServerUpdateListener interface.  I think this puts a bunch of related and coupled logic all in one place.